capactio/capact
# Automate release

## Description

Changes proposed in this pull request:

- Add workflow for releasing Capact
- Use [`gren`](https://github.com/github-tools/github-release-notes) for release note generation

## Testing

You would need to clone, change a few things and push it to your main brach

1. Create an GitHub Environment named "Release" with two secrets:
   - `GH_PAT` - personal access token with permissions to commit to the repository
   - `GCS_CREDS` - base64 encoded GCP credentials in JSON format for GCS write access
2. In `.goreleaser.yml`
   - Change `release.github.owner` to your username
   - Change `blobs[0].bucket` to a test bucket (you can use `capactio-release-testing` I used)
   - Change `dockers[0].image_templates` to point to your GHCR
3. I also commented out most of the `.github/workflows/branch-build.yaml`, so I didn't have to deal with the images pushed there.
4. In `hack/release-charts.sh` change `CAPACTIO_OFFICIAL_BUCKET` env to a test bucket (it must have a valid Helm repo `index.yaml`, you can use `capactio-release-testing` I used).

With these changes, you should be able to test the workflow. Trigger it from `main` and provide eg. `0.5.0` as the version. It should create a release commit on `main`, create a release branch `release-0.5`, push a commit with the Capact image SHA there and tag it. Also, a GitHub release should be created (the release notes will be empty, as your fork most probably does not have any merged PRs/issues).

You can also test a patch release scenario, by triggering the workflow from the `release-0.5` branch and providing `0.5.1` version.

My runs:
- `0.5.0` release - https://github.com/Trojan295/capact/runs/3423994881?check_suite_focus=true
- `0.5.1` release - https://github.com/Trojan295/capact/runs/3424068496?check_suite_focus=true

Example release notes generated by [`gren`](https://github.com/github-tools/github-release-notes) - https://github.com/Trojan295/capact/releases/tag/untagged-a04325e324cfb3cb0597

## Related issue(s)
Resolves https://github.com/capactio/capact/issues/444
